### PR TITLE
Docker Support

### PR DESCRIPTION
**Pull request by Anunayj**
Added Docker support! Also added Examples on how to run it. Might wanna run your own DockerHub repo. Current one should always be up with master with delay of about ~1hr.

### CODE REVIEW

**Comment by Anunayj** [comment]
Docker Image is ~46 MB (Compressed) and ~108MB (Uncompressed), golang is not included in the image.

**Comment by Anunayj** [comment]
Some Useful gists you may wanna add to Readme.md
https://gist.github.com/Anunayj/fa03887a368f5724f6a26ec2a8396a7f
https://gist.github.com/Anunayj/4b75d5cefac875ba51d830d57785c44c

File: README.md [hunk 1]
```diff
@@ -117,6 +117,31 @@ If you use `-skip-dnssec`, letsdane will use the Authenticated Data flag.
 
 I wanted to try DANE, but no browser currently supports it. It may still be a long way to go for browser support, but if you want to try it now you can!
 
+
+## Docker Usage
+
+### Docker CLI
```
Comment (buffrr) [inline README.md:123]: You can put Docker usage after Handshake section so it's easier to notice. For now you can instruct users to build their own image using `docker build -t letsdane .` and use that in the docker run example (i'm planning to maintain images with tagged versions on docker hub but not at the moment).

File: README.md [hunk 2]
```diff
@@ -117,6 +117,31 @@ If you use `-skip-dnssec`, letsdane will use the Authenticated Data flag.
...
+
+### Docker CLI
+
+```bash
+docker run -dp 8080:8080 \
+  -v "$(pwd)"/dane:/root/.letsdane \
+  --restart unless-stopped \
+  anunayj/letsdane -verbose
+```
+### Docker-compose
```
Comment (buffrr) [inline README.md:131]: It's easy to figure out how to use docker-compose based on the docker run example so there is no need for this section. it's better to keep the readme short

**Review by buffrr — CHANGES_REQUESTED** [review]
Great work! The reademe needs a few changes but the Dockerfile looks good so far.

**Comment by Anunayj** [comment]
Reorganized The Readme, and made a small change with the docker run example, instead of listening on `0.0.0.0` it should listen on localhost. I felt it was safer and akin to how letsdane binary works.

**Review by buffrr — APPROVED** [review]
Looks good! Yeah there is no need for the proxy to listen on 0.0.0.0 by default (we should probably change that in the binary too to listen on `127.0.0.1:8080` instead of `:8080`).

**Pull request merged**